app/javascript/bundles/RestaurantsApp/components/Restaurants.jsx

```diff
@@ -15,13 +15,22 @@ export default class Restaurants extends React.Component {
     static propTypes = {
         restaurants: PropTypes.array.isRequired,
         onRestaurantsChange: PropTypes.func.isRequired,
+        onRestaurantClick: PropTypes.func.isRequired,
+        onRestaurantEnter: PropTypes.func.isRequired,
+        onRestaurantLeave: PropTypes.func.isRequired,
     };
 
 
     render() {
         const restaurantList = this.props.restaurants.map((restaurant) => {
             return (
-                <div key={restaurant.id} className="restaurant-card-wrapper">
+                <div
+                    key={restaurant.id}
+                    className="restaurant-card-wrapper"
+                    onClick={() => this.props.onRestaurantClick(restaurant)}
+                    onMouseEnter={() => this.props.onRestaurantEnter(restaurant)}
+                    onMouseLeave={() => this.props.onRestaurantLeave(restaurant)}
```

Comment:
I'm guessing this is used for showing/hiding some elements?
This is something I would try to do with only CSS as much as possible, but it's not imperative.